pennylane_lightning_gpu/src/util/MPIManager.hpp

```diff
@@ -424,6 +424,15 @@ class MPIManager final {
                                      this->getComm()));
     }
 
+    template <typename T>
+    void Reduce(T *sendBuf, T *recvBuf, size_t length, size_t root,
+                const std::string &op_str) {
+        MPI_Datatype datatype = getMPIDatatype<T>();
+        MPI_Op op = getMPIOpType(op_str);
+        PL_MPI_IS_SUCCESS(MPI_Reduce(sendBuf, recvBuf, length, datatype, op,
+                                     root, this->getComm()));
+    }
```

Comment:
Do we want to use raw pointers with this? Could it match the other functions in accepting vectors or similar structures, or is that not an option.

Reply:
Done! Template it with `DataBuffer`.